src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -3353,6 +3353,42 @@ void mvnw(Register Rd, Register Rm,
   INSN(sve_whilelsw, 0b111, 0);
 #undef INSN
 
+private:
+
+  void encode_cvtf_T (SIMD_RegVariant& T_dst, SIMD_RegVariant& T_src) {
+    assert(T_src != B && T_dst != B &&
+           T_src != Q && T_dst != Q, "invalid register variant");
+    if (T_dst != D) {
+      assert(T_dst <= T_src, "invalid register variant");
+    } else {
+      assert(T_src != H, "invalid register variant");
+    }
+    if (T_dst == S && T_src == D) {
+      T_dst = D;
+      T_src = S;
+    } else if (T_dst == D && T_src == S) {
+      T_dst = D;
+      T_src = B;
```

Comment:
This looks confusing. I understand that it's just 0b00, but using B here is really confusing. Maybe using 0b00 and 0b11 directly to align with Arm ARM opc/opc2?

Reply:
Arm uses confusing encoding inst. If we use 0b00 / 0b11 ,  we can not get a concise macroassmbler here ( for example. other developer can not use this macro). In fact, `gcc` uses switch..case to convert macro assmbler into Arm's funny encoding.

Reply:
If we use an `encoding function `(like `switch..case` in `gcc`) instead of `convert function`(`encode_cvtf_T`),  source codes here will be a little complicated. :-)

Reply:
I agree with @nsjian that swapping "T_dst" and "T_src" is confusing although this can generate the same final result.  As I understand that this function is used to generate two opcs from the dst and src type, I'd like not using the swap logic. Thanks!